flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/slotpool/SchedulerImplTest.java

```diff
@@ -0,0 +1,155 @@
+import org.apache.flink.runtime.jobmanager.scheduler.DummyScheduledUnit;
+import org.apache.flink.runtime.jobmaster.LogicalSlot;
+import org.apache.flink.runtime.jobmaster.SlotRequestId;
+import org.apache.flink.runtime.resourcemanager.SlotRequest;
+import org.apache.flink.runtime.resourcemanager.utils.TestingResourceManagerGateway;
+import org.apache.flink.runtime.taskexecutor.slot.SlotOffer;
+import org.apache.flink.runtime.taskmanager.LocalTaskManagerLocation;
+import org.apache.flink.runtime.taskmanager.TaskManagerLocation;
+import org.apache.flink.util.ExceptionUtils;
+import org.apache.flink.util.TestLogger;
+
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import static org.apache.flink.runtime.jobmaster.slotpool.AvailableSlotsTest.DEFAULT_TESTING_PROFILE;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+/**
+ * Tests for the {@link SchedulerImpl}.
+ */
+public class SchedulerImplTest extends TestLogger {
+
+	private static final Time timeout = Time.seconds(1L);
+
+	@ClassRule
+	public static final TestingComponentMainThreadExecutor.Resource EXECUTOR_RESOURCE =
+		new TestingComponentMainThreadExecutor.Resource(10L);
+
+	private final TestingComponentMainThreadExecutor testMainThreadExecutor =
+		EXECUTOR_RESOURCE.getComponentMainThreadTestExecutor();
+
+	private TaskManagerLocation taskManagerLocation;
+	private SimpleAckingTaskManagerGateway taskManagerGateway;
+	private TestingResourceManagerGateway resourceManagerGateway;
+	private SlotPoolBuilder slotPoolBuilder;
+
+	@Before
+	public void setUp() throws Exception {
+		taskManagerLocation = new LocalTaskManagerLocation();
+		taskManagerGateway = new SimpleAckingTaskManagerGateway();
+		resourceManagerGateway = new TestingResourceManagerGateway();
+		slotPoolBuilder = new SlotPoolBuilder(testMainThreadExecutor.getMainThreadExecutor())
+			.setResourceManagerGateway(resourceManagerGateway);
+	}
+
+	@Test
+	public void testAllocateSlot() throws Exception {
+		CompletableFuture<SlotRequest> slotRequestFuture = new CompletableFuture<>();
+		resourceManagerGateway.setRequestSlotConsumer(slotRequestFuture::complete);
+
+		try (SlotPoolImpl slotPool = createAndSetUpSlotPool()) {
+			testMainThreadExecutor.execute(() -> slotPool.registerTaskManager(taskManagerLocation.getResourceID()));
+
+			Scheduler scheduler = new SchedulerImpl(LocationPreferenceSlotSelectionStrategy.createDefault(), slotPool);
+			scheduler.start(testMainThreadExecutor.getMainThreadExecutor());
+
+			SlotRequestId requestId = new SlotRequestId();
+			CompletableFuture<LogicalSlot> future = testMainThreadExecutor.execute(() -> scheduler.allocateSlot(
+				requestId,
```

Comment:
looks to me `requestId` is not further used.
This allocation code is used in both test cases. Could we have an `allocateSlot()` method?